### PR TITLE
Refactor external models: DPOT, VFGN, TopoDiff, mesh_reduced

### PR DESCRIPTION
**Pull request by mnabian**
# PhysicsNeMo Pull Request

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/physicsnemo/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/physicsnemo/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/physicsnemo/issues) is linked to this pull request.
- [ ] If I am implementing a new model or modifying any existing model, I have followed the [Models Implementation Coding Standards](https://github.com/NVIDIA/physicsnemo/wiki/Coding-standards-for-models-implementation).

## Review Process

**All PRs are reviewed by the PhysicsNeMo team before merging.**

Depending on which files are changed, GitHub may automatically assign a maintainer for review.

We are also testing AI-based code review tools (e.g., Greptile), which may add automated comments with a confidence score.
This score reflects the AI’s assessment of merge readiness and is **not** a qualitative judgment of your work, nor is
it an indication that the PR will be accepted / rejected.

AI-generated feedback should be reviewed critically for usefulness.
You are not required to respond to every AI comment, but they are intended to help both authors and reviewers.
Please react to Greptile comments with 👍 or 👎 to provide feedback on their accuracy.

### CODE REVIEW

File: physicsnemo/models/mesh_reduced/mesh_reduced.py [hunk 1]
```diff
@@ -14,307 +14,488 @@
...
-            batch_pivotal = nodes_index.repeat_interleave(
-                torch.tensor([len(position_pivotal)] * graph.batch_size).to(x.device)
+        # Build batched positional tensors and batch ids for KNN interpolation
+        if (pyg is not None) and isinstance(graph, pyg.data.Data):
+            batch_mesh = graph.batch
+            batch_size = (
+                int(batch_mesh.max().item()) + 1 if batch_mesh.numel() > 0 else 1
             )
+        else:
+            raise ValueError(f"Unsupported graph type: {type(graph)}")
```
[the remark this answers is not in the document]
Comment (mnabian) [reply physicsnemo/models/mesh_reduced/mesh_reduced.py]: This comment does not apply to the most recent version of changes.

File: physicsnemo/models/mesh_reduced/mesh_reduced.py [hunk 2]
```diff
@@ -14,307 +14,488 @@
...
+        r"""Decode pivotal features back to mesh space.
 
-            x, _, _, _ = self.knn_interpolate(
-                x=x,
-                pos_x=position_pivotal_batch,
-                pos_y=position_mesh_batch,
-                batch_x=batch_pivotal,
-                batch_y=batch_mesh,
+        Parameters
+        ----------
```
[the remark this answers is not in the document]
Comment (mnabian) [reply physicsnemo/models/mesh_reduced/mesh_reduced.py]: Belongs to an older commit and is not applicable anymore.

File: physicsnemo/models/mesh_reduced/mesh_reduced.py [hunk 3]
```diff
@@ -14,307 +14,412 @@
...
 import torch
+from torch import Tensor
 
-from physicsnemo.core.version_check import check_version_spec
+from physicsnemo.core.version_check import require_version_spec
 from physicsnemo.models.meshgraphnet.meshgraphnet import MeshGraphNet
-from physicsnemo.nn.module.gnn_layers.graph_types import (
-    PYG_AVAILABLE,
-    GraphType,  # noqa
-)
```
Comment (coreyjadams) [inline physicsnemo/models/mesh_reduced/mesh_reduced.py:26]: Can we pivot this to the newer OptionalImport syntax?  I can do it if you like.

File: physicsnemo/models/mesh_reduced/temporal_model.py [hunk 4]
```diff
@@ -14,6 +14,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from typing import Optional
```
Comment (coreyjadams) [inline physicsnemo/models/mesh_reduced/temporal_model.py]: Prefer `|` over Optional?

File: physicsnemo/models/topodiff/diffusion.py [hunk 5]
```diff
@@ -20,11 +20,28 @@
...
+        ----------
+        n_steps : int, optional, default=1000
+            Number of diffusion steps.
+        min_beta : float, optional, default=1e-4
+            Minimum beta in the linear schedule.
+        max_beta : float, optional, default=0.02
+            Maximum beta in the linear schedule.
+        device : str, optional, default="cpu"
+            Target device string for tensors.
+        """
```
Comment (coreyjadams) [inline physicsnemo/models/topodiff/diffusion.py:44]: @CharlelieLrt Is there an easy way to move this on to the new diffusion utilities?

**Review by coreyjadams — APPROVED** [review]
Over all looks fine.  I know these external models are going with a slightly lower standard.  I left a few comments but nothing is mandatory, really, on these.

File: physicsnemo/models/mesh_reduced/mesh_reduced.py (hunk 3 above)
Comment (mnabian) [reply physicsnemo/models/mesh_reduced/mesh_reduced.py:26]: Switched to OptionalImport

File: physicsnemo/models/mesh_reduced/temporal_model.py (hunk 4 above)
Comment (mnabian) [reply physicsnemo/models/mesh_reduced/temporal_model.py]: Sure, switched to |

File: physicsnemo/models/topodiff/diffusion.py (hunk 5 above)
Comment (mnabian) [reply physicsnemo/models/topodiff/diffusion.py:44]: How about we optionally pursue this in a separate PR?
